### PR TITLE
ML2 baremetal - IronicNeutronAgent CRDs

### PR DESCRIPTION
**Pull request by hjensas**
Adds the CRDs and controller to deploy ML2 networking-baremetal "ironic-neutron-agent". 
The Ironic Neutron Agent is "standalone", i.e not deployed by the "ironics" umbrella.

Example:
```
---
apiVersion: ironic.openstack.org/v1beta1
kind: IronicNeutronAgent
metadata:
  name: ironic-neutron-agent
  namespace: openstack
spec:
  serviceUser: ironic
  rabbitMqClusterName: rabbitmq
  containerImage: quay.io/tripleomastercentos9/openstack-ironic-neutron-agent:current-tripleo
  passwordSelectors:
    service: IronicPassword
  replicas: 2
  secret: osp-secret
```

### CODE REVIEW

File: api/v1beta1/ironicneutronagent_types.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2023.
```
Comment (juliakreger) [inline api/v1beta1/ironicneutronagent_types.go:2]: Umm... shouldn't that include who holds it? :)

File: config/crd/bases/ironic.openstack.org_ironicneutronagents.yaml [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                  this service.
+                type: object
+              passwordSelectors:
+                default:
+                  service: IronicPassword
+                description: PasswordSelectors - Selectors to identify the ServiceUser
+                  password and TransportURL from the Secret
+                properties:
+                  database:
+                    default: IronicDatabasePassword
```
Comment (juliakreger) [inline config/crd/bases/ironic.openstack.org_ironicneutronagents.yaml:102]: ... Why does this part of the CRD need a password?

**Review by juliakreger — CHANGES_REQUESTED** [review]
I think there is some copy-pasta in this, in regards to the database. Which seems wrong to me, given the service doesn't use it.

File: api/v1beta1/ironicneutronagent_types.go (hunk 1 above)
Comment (hjensas) [reply api/v1beta1/ironicneutronagent_types.go]: Agreed. I copied this from other files. It seems missing copyright holder is an issue in _a lot_ of places across the github.com/openstack-k8s-operators repos.

I've updated the files in this PR, but we should follow up on existing files.

File: config/crd/bases/ironic.openstack.org_ironicneutronagents.yaml (hunk 2 above)
Comment (hjensas) [reply config/crd/bases/ironic.openstack.org_ironicneutronagents.yaml]: The CRD is generated from code, I re-used PasswordSelectors from ironic_types.go. (https://github.com/openstack-k8s-operators/ironic-operator/blob/master/api/v1beta1/ironic_types.go#L118)
Updated, there is a separate IronicNeutronAgentPasswordSelector with just the service now.

**Comment by hjensas** [comment]
> I think there is some copy-pasta in this, in regards to the database. Which seems wrong to me, given the service doesn't use it.

Done.

**Comment by hjensas** [comment]
ooops, did not mean to close this.

**Pull request closed without merging**